ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionQueryBuilder.java

```diff
@@ -275,137 +250,105 @@ String build() {
-    deltas.forEach(delta -> query.append("partition (file_name='")
-        .append(delta.getPath().getName()).append("')"
-            + " location '").append(delta.getPath()).append("' "));
-  }
-
-
-  private void buildSelectClauseForInsert(StringBuilder query) {
-    // Need list of columns for major crud, mmmajor partitioned, mmminor
-    List<FieldSchema> cols;
-    if (CompactionType.REBALANCE.equals(compactionType) ||
-        CompactionType.MAJOR.equals(compactionType) && (!insertOnly || sourcePartition != null) ||
-        CompactionType.MINOR.equals(compactionType) && insertOnly) {
-      if (sourceTab == null) {
-        return; // avoid NPEs, don't throw an exception but skip this part of the query
-      }
-      cols = sourceTab.getSd().getCols();
-    } else {
-      cols = null;
-    }
-    switch (compactionType) {
-      case REBALANCE: {
-        query.append("0, t2.writeId, t2.rowId DIV CEIL(numRows / ")
-            .append(numberOfBuckets)
-            .append("), t2.rowId, t2.writeId, t2.data from (select ")
-            .append("count(ROW__ID.writeId) over() as numRows, ");
-        if (StringUtils.isNotBlank(orderByClause)) {
-          // in case of reordering the data the writeids cannot be kept.
-          query.append("MAX(ROW__ID.writeId) over() as writeId, row_number() OVER (")
-            .append(orderByClause);
-        } else {
-          query.append("ROW__ID.writeId as writeId, row_number() OVER (order by ROW__ID.writeId ASC, ROW__ID.bucketId ASC, ROW__ID.rowId ASC");
-        }
-        query.append(") - 1 AS rowId, NAMED_STRUCT(");
-        for (int i = 0; i < cols.size(); ++i) {
-          query.append(i == 0 ? "'" : ", '").append(cols.get(i).getName()).append("', `")
-              .append(cols.get(i).getName()).append("`");
-        }
-        query.append(") as data");
-        break;
-      }
-      case MAJOR: {
-        if (insertOnly) {
-          if (sourcePartition != null) { //mmmajor and partitioned
-            appendColumns(query, cols, false);
-          } else { // mmmajor and unpartitioned
-            query.append("*");
-          }
-        } else {
-          query.append(
-              "validate_acid_sort_order(ROW__ID.writeId, ROW__ID.bucketId, ROW__ID.rowId), "
-                  + "ROW__ID.writeId, ROW__ID.bucketId, ROW__ID.rowId, ROW__ID.writeId, "
-                  + "NAMED_STRUCT(");
-          appendColumns(query, cols, true);
-          query.append(") ");
-        }
-        break;
-      }
-      case MINOR: {
-        if (insertOnly) {
-          appendColumns(query, cols, false);
-        } else {
-          query.append(
-              "`operation`, `originalTransaction`, `bucket`, `rowId`, `currentTransaction`, `row`");
-        }
-        break;
-      }
-    }
+    deltas.forEach(
+        delta -> query.append("partition (file_name='").append(delta.getPath().getName()).append("')" + " location '")
+            .append(delta.getPath()).append("' "));
```

Comment:
Please avoid the concatenation : append("')" + " location '")

Reply:
Fixed it.